cranelift/interpreter/src/value.rs

```diff
@@ -808,4 +819,97 @@ impl DataValueExt for DataValue {
     fn swap_bytes(self) -> ValueResult<Self> {
         unary_match!(swap_bytes(&self); [I16, I32, I64, I128])
     }
+
+    fn iter_lanes(&self, ty: Type) -> DataValueIterator {
+        DataValueIterator::new(self, ty)
+    }
+}
+
+/// Iterator for DataValue's
+pub struct DataValueIterator {
+    ty: Type,
+    v: SimdVec<DataValue>,
+    idx: usize,
+}
+
+impl DataValueIterator {
+    fn new(dv: &DataValue, ty: Type) -> Self {
+        match extractlanes(dv, ty) {
+            Ok(v) => return Self { ty, v, idx: 0 },
+            Err(_) => {
```

Comment:
Lets propagate up the error from `extractlanes`. It might help debug something in the future, instead of silently having an empty iterator.

Reply:
Good idea, fixed with:
https://github.com/ms140569/wasmtime/commit/815c6b3e285a9661be28798d61848b0302bf2797